File: front/src/api/render-task.ts
import { request } from './request'

export interface RenderTaskVO {
  id?: number
  taskId: string
  taskType: 'SHOT_FRAME' | 'SHOT_VIDEO' | 'ASSET_IMAGE' | 'GROUP_SERIAL' // GROUP_SERIAL is legacy history only
  taskName: string
  dramaId?: number
  dramaTitle?: string
  episodeId?: number
  episodeName?: string
  sceneId?: number
  shotId?: number
  shotNo?: number
  shotGroupId?: number
  assetType?: string
  assetId?: number
  assetName?: string
  assetSlot?: string
  providerId?: number
  providerName?: string
  modelCode?: string
  status: 'QUEUED' | 'RENDERING' | 'SUCCESS' | 'FAILED' | 'CANCELLED'
  progress: number
  currentNode?: string
  prompt?: string
  negativePrompt?: string
  referenceImages?: string
  outputUrl?: string
  lastFrameUrl?: string
  errorMessage?: string
  errorDetail?: string
  submitTime?: string
  startTime?: string
  finishTime?: string
  costMs?: number
}

export interface RenderTaskQuery {
  current?: number
  size?: number
  dramaId?: number
  episodeId?: number
  taskType?: string
  status?: string
  keyword?: string
  startTime?: string
  endTime?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  size: number
  current: number
  pages: number
}

export const renderTaskApi = {
  // 获取当前实时活跃任务列表 (从 Redis 缓存)
  getActiveTasks() {
    return request<RenderTaskVO[]>({
      url: '/render/tasks/active',
      method: 'get'
    })
  },

  // 获取当前活跃任务总数 (Redis)
  getActiveCount() {
    return request<number>({
      url: '/render/tasks/active/count',
      method: 'get'
    })
  },

  // 根据任务 ID 查询详情
  getTaskById(taskId: string) {
    return request<RenderTaskVO>({
      url: `/render/tasks/${taskId}`,
      method: 'get'
    })
  },

  // 分页查询历史归档记录 (走 MySQL)
  getHistoryTasks(params: RenderTaskQuery) {
    return request<PageResult<RenderTaskVO>>({
      url: '/render/tasks/history',
      method: 'get',
      params
    })
  },

  // 主动取消任务
  cancelTask(taskId: string) {
    return request<boolean>({
      url: `/render/tasks/${taskId}/cancel`,
      method: 'post'
    })
  }
}

File: src/main/java/com/astra/freyja/entity/RenderTask.java
package com.astra.freyja.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDateTime;

/**
 * 渲染任务与历史归档持久化实体。
 */
@Data
@EqualsAndHashCode(callSuper = true)
@TableName("render_task")
public class RenderTask extends BaseEntity {

    /** 业务任务唯一标识 (如 RENDER_1710000000) */
    private String taskId;

    /** 任务类型: SHOT_FRAME(关键帧), SHOT_VIDEO(视频), GROUP_SERIAL(历史数据，仅用于兼容读取), ASSET_IMAGE(资产生图) */
    private String taskType;

    /** 任务展示名称 (如: [第1集 S1 镜头名] 首帧渲染) */
    private String taskName;

    /** 关联短剧 ID */
    private Long dramaId;

    /** 关联剧集 ID */
    private Long episodeId;

    /** 关联场次 ID */
    private Long sceneId;

    /** 关联分镜 ID */
    private Long shotId;

    /** 关联镜头组 ID */
    private Long shotGroupId;

    /** 资产类型: CHARACTER/SCENE/PROP */
    private String assetType;

    /** 资产目标 ID */
    private Long assetId;

    /** 资产图片槽位 */
    private String assetSlot;

    /** AI 提供商 ID */
    private Long providerId;

    /** AI 提供商名称 */
    private String providerName;

    /** AI 模型代码 */
    private String modelCode;

    /** 任务状态: QUEUED(排队中)/RENDERING(渲染中)/SUCCESS(成功)/FAILED(失败)/CANCELLED(已取消) */
    private String status;

    /** 当前进度百分比 (0-100) */
    private Integer progress;

    /** 当前渲染节点/阶段描述 */
    private String currentNode;

    /** 渲染正向提示词快照 */
    private String prompt;

    /** 渲染负向提示词快照 */
    private String negativePrompt;

    /** 参考图列表 (JSON Array) */
    private String referenceImages;

    /** 渲染产物 MinIO 访问 URL */
    private String outputUrl;

    /** 抽取尾帧 MinIO URL */
    private String lastFrameUrl;

    /** 异常简要信息 */
    private String errorMessage;

    /** 错误堆栈 */
    private String errorDetail;

    /** 任务提交时间 */
    private LocalDateTime submitTime;

    /** 任务开始执行时间 */
    private LocalDateTime startTime;

    /** 任务结束时间 */
    private LocalDateTime finishTime;

    /** 总执行耗时(毫秒) */
    private Long costMs;
}
